### DISCUSSION TITLE
Custom Client: set user

### DISCUSSION THREAD

**Discussion by mahescho** [rustdesk/rustdesk-server-pro#304] [Ideas]
Please add an advanced option to set the user for a client. This is the bare minimum to work around the permission problems described here:

https://github.com/rustdesk/rustdesk-server-pro/discussions/295

**Comment by rustdesk** [comment]
ready, https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#preset-user-name--preset-strategy-name

**Comment by mahescho** [comment]
Thanks for the quick fix. Works as expected. In conjunction with the other new advanced settings it's ready for roll out for me.

**Comment by lamanna-dev** [comment]
I cant seem to get this to work. I set preset-user-name=<user> in the override/default settings in the custom client generator, yet when the device is shown in the devices list, it does not have the user assigned - still need to add it manually.
The user already exists in the console.

**Reply by rustdesk** [reply]
Make sure `API` is set correctly, https://github.com/rustdesk/rustdesk-server-pro/issues/267

**Reply by lamanna-dev** [reply]
I had already done that - and can connect to the console/logon from the client. 
Perhaps because I am using a reverse proxy (iis) to connect to port 443 -> 21114 on the server?
is there a specific api address/port that is used so I can test?

**Reply by lamanna-dev** [reply]
Actually, I got it working; that one custom client had http rather than https as the api address and I missed it when trying to work out what was wrong.
